t/25-cache.t

```diff
@@ -0,0 +1,247 @@
+    open(my $tmpfd, '>:raw', $filename);
+    print $tmpfd "\0" x 84 or die($filename);
+    close $tmpfd;
+
+    if ($_ % 2) {
+        log_info "Inserting $_";
+        $sql = "INSERT INTO assets (downloading,filename,size, etag,last_use)
+                VALUES (0, ?, ?, 'Not valid', strftime('%s','now'));";
+        $sth = $dbh->prepare($sql);
+        $sth->bind_param(1, $filename);
+        $sth->bind_param(2, 84);
+        $sth->execute();
+    }
+
+}
+
+OpenQA::Worker::Cache::init($host, $cachedir);
+
+unlike read_log, qr/Deploying DB/, "Cache deploys the database.";
+like read_log, qr/CACHE: Health: Real size: 168, Configured limit: 53687091200/,
+  "Cache limit/size match the expected 100GB/168)";
+unlike read_log, qr/CACHE: Purging non registered.*[13].qcow2/, "Registered assets 1 and 3 were kept";
+like read_log,   qr/CACHE: Purging non registered.*2.qcow2/,    "Asset 2 was removed";
+truncate_log;
+
+$OpenQA::Worker::Cache::limit = 100;
+OpenQA::Worker::Cache::init($host, $cachedir);
+
+like read_log, qr/CACHE: Health: Real size: 84, Configured limit: 100/, "Cache limit/size match the expected 100/84)";
+like read_log, qr/CACHE: removed.*1.qcow2/, "Oldest asset (1.qcow2) removal was logged";
+ok(!-e "$cachedir/1.qcow2", "Oldest asset (1.qcow2) was sucessfully removed");
+truncate_log;
+my $daemon;
+my $mock = Mojolicious->new;
+
+sub start_server {
+    my $serverpid = fork();
+    if ($serverpid == 0) {
+        # setup mock
+
+        $mock->routes->get(
+            '/tests/:job/asset/:type/:filename' => sub {
+                my $c        = shift;
+                my $id       = $c->stash('job');
+                my $type     = $c->stash('type');
+                my $filename = $c->stash('filename');
+                return $c->render(status => 404, text => "Move along, nothing to see here")
+                  if $filename =~ /sle-12-SP3-x86_64-0368-404/;
+                return $c->render(status => 400, text => "Move along, nothing to see here")
+                  if $filename =~ /sle-12-SP3-x86_64-0368-400/;
+                return $c->render(status => 500, text => "Move along, nothing to see here")
+                  if $filename =~ /sle-12-SP3-x86_64-0368-589/;
+                if ($filename =~ /sle-12-SP3-x86_64-0368-589/) {
+                    $c->res->headers->content_length(10);
+                    $c->write(
+                        'Hel' => sub {
+                            my $c = shift;
+                            $c->write('lo!');
+                        });
+                }
+            });
+
+        $mock->routes->get(
+            '/' => sub {
+                my $c = shift;
+                return $c->render(status => 200, text => "server is running");
+            });
+
+        # Connect application with web server and start accepting connections
+
+        $daemon = Mojo::Server::Daemon->new(app => $mock, listen => [$host]);
+        $daemon->start;
+    }
+}
+
+start_server;
```

Comment:
I think this could result in a race condition if the forked test server doesn't bind to the listen socket fast enough. Which can be avoided by waiting a little in the test process.
```perl
use IO::Socket::INET;

sub _port { IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => shift) }

sleep 1 while !_port($port);
```
We use that in [Mojolicious tests](https://github.com/kraih/mojo/blob/master/t/mojo/morbo.t#L44) too.